### PR TITLE
Data Dictionary Markdown Generator

### PR DESCRIPTION
**Pull request by crossan007**
Adds a generator for creating markdown data dictionaries.

usage: ```propel --config-dir=propel datadictionary:generate```

Proof of concept usage here: https://github.com/ChurchCRM/CRM/blob/5be16c7f1016ac117556dab623e5c9efdf7d2065/generated-datadictionary/default.schema.md

### CODE REVIEW

File: src/Propel/Generator/Manager/DataDictionaryManager.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+use Propel\Generator\Model\Column;
+
+/**
+ * Manager for Markdown Data Dictionary
+ *
+ * @author Charles Crossan <crossan007@gmail.com>
+ */
+class DataDictionaryManager extends AbstractManager
+{
+    private $dictionaryColumns = ["Column Name","PHP Name", "PK/FK","Format","Length","Description"];
```
Comment (marcj) [inline src/Propel/Generator/Manager/DataDictionaryManager.php]: you can use a const here.

File: src/Propel/Generator/Manager/DataDictionaryManager.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    {
+      return preg_replace('/[^A-Z|a-z]/','-',$name);
+    }
+    
+    public function build()
+    {
+        $count = 0;
+
+        foreach ($this->getDatabases() as $database) {
+            $MdSyntax = "# Data Dictionary for " .  $database->getName() ."\n";
```
Comment (marcj) [inline src/Propel/Generator/Manager/DataDictionaryManager.php]: variable name start lower case please

File: src/Propel/Generator/Manager/DataDictionaryManager.php [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                $MdSyntax .= "<a name=\"".$this->getAnchorName($tbl->getName())."\"></a>\n## Table: " . $tbl->getName() . "\n";
+                $MdSyntax .= "[Table of Contents](#TOC)\n\n";
+                
+                if ($tbl->getDescription())
+                {
+                  $MdSyntax .= "### Description:\n";
+                  $MdSyntax .= $tbl->getDescription()."\n";
+                }
+                
+               $MdSyntax .= "### Columns:\n";
```
Comment (marcj) [inline src/Propel/Generator/Manager/DataDictionaryManager.php]: indentation

File: src/Propel/Generator/Manager/DataDictionaryManager.php [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+                $MdSyntax .= "|".join("|",array_fill(0,count($this->dictionaryColumns),"---"))."|\n";
+
+                foreach ($tbl->getColumns() as $col) {
+                    $columnRow = [];
+                    $columnRow[0] = $col->getName();
+                    $columnRow[1] = $col->getPhpName();
+                    if (count($col->getForeignKeys()) > 0) {
+                      $columnRow[2] = '';
+                      foreach ($col->getForeignKeys() as $fk)
+                      {
```
Comment (marcj) [inline src/Propel/Generator/Manager/DataDictionaryManager.php]: no new line for `{`

**Comment by crossan007** [comment]
updated generated markdown file based on latest commit to this branch: https://github.com/ChurchCRM/CRM/wiki/default.schema

**Comment by dereuromark** [comment]
New PR https://github.com/propelorm/Propel2/pull/1737

**Pull request closed without merging**